Tests/Integration/tests/Sitemap/AssertUtil.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Presta\SitemapBundle\Tests\Integration\Tests\Sitemap;
+
+use PHPUnit\Framework\Assert;
+
+final class AssertUtil
+{
+    private const BASE_URL = 'http://localhost';
+
+    public static function assertIndex(string $xml)
+    {
+        self::assertIndexContainsSectionLink($xml, 'static');
+        self::assertIndexContainsSectionLink($xml, 'blog');
+        //todo more assertions
+    }
+
+    public static function assertStaticSection(string $xml)
+    {
+        self::assertSectionContainsPath($xml, 'static', '/annotation');
+        self::assertSectionContainsPath($xml, 'static', '/yaml');
+        self::assertSectionContainsPath($xml, 'static', '/xml');
+        //todo more assertions
+    }
+
+    public static function assertBlogSection(string $xml)
+    {
+        self::assertSectionContainsPath($xml, 'blog', '/blog');
+        self::assertSectionContainsPath($xml, 'blog', '/blog/foo');
+        //todo more assertions
+    }
+
+    private static function assertIndexContainsSectionLink(string $xml, string $name)
+    {
+        $loc = preg_quote(sprintf('%s/sitemap.%s.xml', self::BASE_URL, $name), '#');
+        $lastmod = self::approximatedDateAsRegex();
+
+        Assert::assertRegExp(
```

Comment:
Yes I was planning to use DomCrawler, but I will have to perform same assertions the same way in both HTTP & CLI context. So we ensure sitemaps are **exactly** equals

Reply:
@ostrolucky I spent some hours on trying to test sitemaps, but I got annoyed with namespaces import... no way I can test media, did you tried something like this before ?

Reply:
Sorry I'm not sure what are you asking. What do you mean with media and how did your idea to test it look like?

But I don't think so, what I posted is pretty much our whole test suite regarding sitemaps we currently have.

Reply:
The problem I have is that XML parsing with Crawler (or SimpleXML) is failing because of custom namespaces included in media sitemaps.
I did not find a way to assert that my links has images, videos... because of that.
I was wondering if you had any experience with that, because I wanted to test it.

Reply:
I see. No, I don't think I have experience, but I can help you look into it if you want, if you push your failing test case.

Reply:
I removed the assertions on image caption, then everything works, I will keep to this for now...